bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/ChallengeResponse/GuardianChallengeResponse.cs

```diff
@@ -0,0 +1,99 @@
+using ElectionGuard.Decryption.Challenge;
+using ElectionGuard.Decryption.Shares;
+using ElectionGuard.Decryption.Tally;
+using ElectionGuard.ElectionSetup;
+using ElectionGuard.Extensions;
+using ElectionGuard.Guardians;
+
+namespace ElectionGuard.Decryption.ChallengeResponse;
+
+/// <summary>
+/// A challenge for a specific guardian to prove that a decryption share was generated correctly.
+/// </summary>
+public record GuardianChallengeResponse : DisposableRecordBase
+{
+    // TODO: tallyId?
+
+    /// <summary>
+    /// The object id of the guardian.
+    /// </summary>
+    public string GuardianId { get; init; }
+
+    /// <summary>
+    /// The sequence order of the guardian.
+    /// </summary>
+    public ulong SequenceOrder { get; init; }
+
+    /// <summary>
+    /// The offset used to generate the commitment.
+    /// </summary>
+    public ElementModP CommitmentOffset { get; init; }
+
+    public TallyChallengeResponse Tally { get; init; }
+
+    public List<BallotChallengeResponse> Ballots { get; init; }
+
+    public GuardianChallengeResponse(
+        IElectionGuardian guardian,
+        ElementModP commitmentOffset,
+        TallyChallengeResponse tally,
+        List<BallotChallengeResponse>? ballots)
+    {
+        GuardianId = guardian.GuardianId;
+        SequenceOrder = guardian.SequenceOrder;
+        CommitmentOffset = new(commitmentOffset);
+        Tally = tally;
```

Comment:
Should this be a new like in the ctor below?

Reply:
theoretically it shouldn't be given that this object is a container and is taking ownership of the Tally & ballots lists, whereas the constructor below is an explicit copy constructor so we duplicate the object. That said, i know this has caused some problems in the past. what do you think?